project/npda/templates/partials/visit_form_tab.html

```diff
@@ -4,114 +4,91 @@
        role="tab"
        class="basis-full md:basis-1/3 tab font-bold mb-2 text-black border-b-4 hover:border-b-rcpch_pink border-b-rcpch_light_blue_tint3 border-solid border-0 bg-rcpch_light_blue"
        value="{{ label|lowerify }}"
-       aria-label="{{ label }} {% if categories|categories_have_errors:form %}❗{% endif %}"
+       aria-label="{{ label }} {% if errors %}❗{% endif %}"
        {% if checked %}checked{% endif %} />
 <div role="tabpanel" class="tab-content order-1 basis-full rounded-none">
   <div class="flex align-middle items-center bg-gray-100 p-2 m-2">
     <span class="mr-2 text-nowrap"><i class="fa-solid fa-list"></i> Skip to section: </span>
     <ul class="flex gap-2 flex-wrap my-2">
-      {% for field_category in categories|split_by_comma %}
-        <li class="inline-block p-2 font-mono bg-{{ field_category|colour_for_category }} hover:shadow-md hover:underline">
-          <a href="#{{ field_category|cut:" " }}_anchor">
-            {% if field_category|category_has_errors:form %}
+      {% for category in categories %}
+        <li class="inline-block p-2 font-mono bg-{{ category.colour }} hover:shadow-md hover:underline">
+          <a href="#{{category.anchor}}">
+            {% if category.errors %}
               <span class="fa-stack text-rcpch_red">
                 <i class="fa-circle fa-stack-1x fas"></i>
                 <i class="fa-exclamation fa-stack-1x fa-inverse fas"></i>
               </span>
             {% endif %}
-            {{ field_category }}
+            {{ category.name }}
           </a>
         </li>
       {% endfor %}
     </ul>
   </div>
-  {% for field_category in categories|split_by_comma %}
```

Comment:
This diff looks horrendous but it's the same code with one less level of nesting as we can now remove the `with` for background colour